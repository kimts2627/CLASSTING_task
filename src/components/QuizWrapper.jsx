import axios from "axios";
import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import Footer from "./Footer";
import QuizMain from "./QuizMain";
import QuizPage from "./QuizPage";

const QuizWrapper = () => {
  const quiz = useSelector((state) => state.quiz.quiz);
  const currentQuiz = useSelector((state) => state.quiz.currentQuiz);
  return (
    <div className="relative w-full max-w-md h-full flex flex-col items-center z-10">
      {quiz ? <QuizPage page={currentQuiz} /> : <QuizMain />}
      <Footer />
    </div>
  );
};

export default QuizWrapper;
