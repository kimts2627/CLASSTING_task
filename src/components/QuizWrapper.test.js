import React from "react";
import QuizWrapper from "./QuizWrapper";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { store } from "../index";
import { Provider } from "react-redux";

configure({ adapter: new Adapter() });
let component = null;

describe("QuizWrapper", () => {
  it("컴포넌트가 정상적으로 렌더링 되어야 함", () => {
    component = shallow(
      <Provider store={store}>
        <QuizWrapper />
      </Provider>
    );
    expect(component.length).toBe(1);
  });

  it("스냅샷과 결과물이 일치해야 함", () => {
    expect(component).toMatchSnapshot();
  });
});
