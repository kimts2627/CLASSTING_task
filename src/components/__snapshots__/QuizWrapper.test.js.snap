// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizWrapper 스냅샷과 결과물이 일치해야 함 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <QuizWrapper />
</ContextProvider>
`;
